### ISSUE TITLE
Delete an object and its dependencies

### ISSUE THREAD

**Issue by LaraEngebretson** [AtomicGameEngine/AtomicGameEngine#235]
To keep the Project folder clean, it would be handy to be able to delete an object's material and its texture(s) when deleting the object from the project. This would be optional, since there is a potential for other objects to use the same material/textures.

**Comment by Alan-FGR** [comment]
I don't agree with this, is there any engine that does that? Maybe it could tell the user that a given asset has no 'users' like Blender but that would require keeping track of all assets in all scenes what's totally bug prone and could cause tons of problems like assets being deleted accidentally. I don't recommend it. To keep distributable size down just check at build time which assets aren't used.
@mattbenic input&close
